Source/JavaScriptCore/jit/OperationResult.h

```diff
@@ -138,6 +141,117 @@ ALWAYS_INLINE T makeOperationResult(Scope& scope, T result)
     RELEASE_AND_RETURN(scope, result);
 }
 
+#else // USE(JSVALUE64)
+
+enum class ExceptionOperationResultTag : uint64_t { };
+
+struct ExceptionOperationResultVoid {
+    Exception* exception;
+};
+
+template<typename T>
+using ExceptionOperationResult = std::conditional_t<(std::is_same_v<T, void>), ExceptionOperationResultVoid, ExceptionOperationResultTag>;
+
+template<typename OperationType>
+concept OperationHasResult = FunctionTraits<OperationType>::hasResult && !std::is_same_v<typename FunctionTraits<OperationType>::ResultType, ExceptionOperationResult<void>>;
+
+// Note we need this because e.g. `requires (!OperationHasResult<int>)` will pass the requirement as the SFINAE in `OperationHasResult<int>`
+// means the concept returns false and the `!` makes it true.
+template<typename OperationType>
+concept OperationIsVoid = !FunctionTraits<OperationType>::hasResult || std::is_same_v<typename FunctionTraits<OperationType>::ResultType, ExceptionOperationResult<void>>;
+
+template<typename T>
+struct IsExceptionOperationResult : std::false_type { };
+
+template<>
+struct IsExceptionOperationResult<ExceptionOperationResultTag> : std::true_type { };
+
+template<>
+struct IsExceptionOperationResult<ExceptionOperationResultVoid> : std::true_type { };
+
+template<typename T>
+concept isExceptionOperationResult = IsExceptionOperationResult<T>::value;
+
+// Note: We always return an exception in a register if the operation's return type is void on all platforms (assuming it throws).
+template<typename T>
+using OperationReturnType = std::conditional_t<(assert_not_operation_signature<T> && (std::is_same_v<T, void> || canMakeExceptionOperationResult<T>)), ExceptionOperationResult<T>, T>;
+
+static ALWAYS_INLINE ExceptionOperationResultTag asExceptionResultBase(uint32_t value, Exception* exception)
+{
+    uint32_t v = bitwise_cast<uint32_t>(value);
+    return bitwise_cast<ExceptionOperationResultTag>(
+        static_cast<long long>(v)
+        | (static_cast<long long>(reinterpret_cast<uint32_t>(exception)) << 32));
+}
+
+template<typename T>
+static ALWAYS_INLINE ExceptionOperationResultTag asExceptionResult(T value, Exception* exception)
+requires (sizeof(T) == sizeof(uint32_t))
```

Comment:
I was putting the `requires` clauses between the template and the signature. It seems like we should be consistent or change the other cases to match this.